packages/protect/src/ffi/model-helpers.ts

```diff
@@ -378,18 +499,40 @@ export async function bulkEncryptModels<T extends Record<string, unknown>>(
     })),
   )
 
-  // Make a single FFI call for all fields
   const encryptedData = await handleMultiModelBulkOperation(
     bulkEncryptPayload,
     (items) => encryptBulk(client, items),
     keyMap,
   )
 
-  // Reconstruct models
-  return models.map((_, modelIndex) => ({
-    ...otherFields[modelIndex],
-    ...nullFields[modelIndex],
-    ...Object.fromEntries(
+  // Helper function to set a nested value
+  const setNestedValue = (
```

Comment:
Is this the same as the `setNestedValue` function above? If yes, it might be worth extracting this function and sharing it.